### PR TITLE
LRDOCS-2687 upgrading your app upgrade and verifier processes tutorial

### PR DESCRIPTION
**Pull request by mwilliams2014**

### CODE REVIEW

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+## Upgrading Your Upgrade Process to the New Framework
+
+Follow the steps below to migrate your code to the new framework.
+
+1.  If your application has any dependencies, add a dependency on the 
+    `portal-upgrade` module to your `build.gradle` file:
+    
+        provided group: "com.liferay", name: "com.liferay.portal.upgrade", 
+        version: "2.0.0"
```
Comment (jhinkey) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: Will there be cases in which the developer's application does not already use a dependency tool, such as Gradle, Maven, or Ivy? For example, might their application be a traditional Plugins SDK app that doesn't use them? If so, this step should speak to that and give them a link to instructions (e.g., [Adapting to Liferay 7's API with the Code Upgrade Tool](/develop/tutorials/-/knowledge_base/7-0/adapting-to-liferay-7s-api-with-the-code-upgrade-tool) ) for handling such a case.

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+3.  Convert your `upgrade` class into an OSGi component, using the `@Component` 
+    token and implement the `UpgradeStepRegistrator` interface, as shown in the
+    example below:
+    
+        @Component(
+                immediate = true, 
+                service = UpgradeStepRegistrator.class
+        )
+
+        public class CalendarServiceUpgrade implements UpgradeStepRegistrator
```
Comment (jhinkey) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: Note, I think the closing parenthesis should be in the line just above the class declaration.

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        upgrade.processes=
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_0,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_1,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_2
+    
+    Each step between versions was represented by a single class extending
+    `UpgradeProcess`, using a method called `doUpgrade`. This method was 
+    responsible for executing the internal steps to update the database to that 
+    concrete version. A method `getThreshold` is provided also to specify the 
+    schema version where the upgrade starts.
```
Comment (jhinkey) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: Do you mean "A method `getThreshold` **was** provided ..."?

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+
+    In this example, the `com.liferay.calendar.service` module is being upgraded 
+    from version 0.0.1 to version 1.0.0. The changes are produced by a list of 
+    `UpgradeSteps` instances, which in this example contains only one step:
+
+        new com.liferay.calendar.upgrade.v1_0_0.UpgradeCalendarBooking()
+
+    The internal steps defined within the intermediate classes, the former 
+    `UpgradeProcess` class, as they are indeed `UpgradeSteps`, require no change
+    on your part. The new framework will process the steps as they are. The
```
Comment (jhinkey) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: I'm not sure what the first sentence "The internal steps..." means. As I read it, here's how I understand it ...
- The internal steps defined the former `UpgradeProcess` class within the intermediate classes.
- The internal steps are `UpgradeSteps` instances.
- You don't need to change the internal steps.

Can you clarify? Thanks.

**Comment by jhinkey** [comment]
Closing for Mike to incorporate the additional clarifications and resend.

https://issues.liferay.com/browse/LRDOCS-2687

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown (hunk 3 above)
Comment (mdelapenya) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: Yes

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown (hunk 2 above)
Comment (mdelapenya) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: Yes

File: develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown (hunk 1 above)
Comment (mdelapenya) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: If a developer is not using dependency management, he/she has not suffered enough

Comment (mwilliams2014) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: I'll take that as a no :)

Comment (mdelapenya) [inline develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown]: In fact, our legacy plugins SDK was using `Ivy` for dependency management, so those previous version should have no problems when migrating

**Pull request closed without merging**